### PR TITLE
Add operator marketplace to dependencies for CI

### PR DESCRIPTION
**Pull request by aakarshg**

### CODE REVIEW

File: tests/common.sh [hunk 1]
```diff
@@ -18,7 +18,30 @@ function delete_operator {
   kubectl delete -f resources/operator.yaml
 }
 
+function olm_dep {
```
Comment (dustinblack) [inline tests/common.sh]: So technically this is the marketplace, not the OLM (the OLM is a dependency of the marketplace, and it is already on the CI systems).

File: tests/common.sh [hunk 2]
```diff
@@ -18,7 +18,30 @@ function delete_operator {
...
+  kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/01_namespace.yaml
+  kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/02_catalogsourceconfig.crd.yaml
+  kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/03_operatorsource.crd.yaml
+  kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/04_service_account.yaml
+  kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/05_role.yaml
+  kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/06_role_binding.yaml
+  kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/07_upstream_operatorsource.cr.yaml
+  kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/08_operator.yaml
```
Comment (dustinblack) [inline tests/common.sh]: Applying each file technically works, but if they change the structure then we have to maintain against it.

File: tests/common.sh (hunk 1 above)
Comment (aakarshg) [reply tests/common.sh]: ack will update it.

File: tests/common.sh (hunk 2 above)
Comment (aakarshg) [reply tests/common.sh]: it makes sense, i'll try the sparse checkout

File: tests/common.sh [hunk 3]
```diff
@@ -18,7 +18,21 @@ function delete_operator {
   kubectl delete -f resources/operator.yaml
 }
 
+function marketplace_setup {
+  git clone --depth 1 --no-checkout --filter=blob:none   "https://github.com/operator-framework/operator-marketplace/" local_repo
+  cd local_repo
```
Comment (dustinblack) [inline tests/common.sh]: Maybe `pushd local_repo` and then `popd` below instead of `cd ..`

File: tests/common.sh [hunk 4]
```diff
@@ -18,7 +18,21 @@ function delete_operator {
   kubectl delete -f resources/operator.yaml
 }
 
+function marketplace_setup {
+  git clone --depth 1 --no-checkout --filter=blob:none   "https://github.com/operator-framework/operator-marketplace/" local_repo
```
Comment (dustinblack) [inline tests/common.sh]: This still seems to do a full clone, which I thought you were trying to avoid.

**Comment by aakarshg** [comment]
Falling back to the original approach of applying the 8 files files and deleting on exit. The original approach works better for the following reasons:
- No need to clone the repo or a part of it
- Simplifying the task
- If there's any change to directory structure, the builds would fail so we would notice immediately.

**Pull request merged**